content/blog/understanding-event-loop/index.md

```diff
@@ -0,0 +1,102 @@
+---
+title: "Understanding event loop in JavaScript"
+date: "2020-09-16"
+coverImage: "index.png"
+author: "Aman Agrawal"
+tags: ["JavaScript", "event-loop", "call stack", "event queue", "multi-threaded"]
+description: "Learn the basic concepts about the JavaScript Event Loop."
+---
+
+We often heard that JavaScript is a single-threaded programming language, which means it executes all of the instructions line by line in a synchronous manner. Thus since everything works on the main thread, there seems to be no possibility of executing parallel processes in JavaScript.
+
+But now let's assume that we have a function which is taking some time to execute, for example in the below function which is having a loop over 10k times, the `console.log()` would be executed once the loop is over and in the meantime, our UI interaction with the browser would be interrupted.
+
+```JavaScript
+function someTimeTakingFunc() {
+  for(var i = 0; i < 10; i++) {
```

Comment:
loops runs 10 times but above it's said it runs 10k times

Reply:
Sure , will chnage it